### DISCUSSION TITLE
v1.78.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#4079] [Announcements]
# v1.78.0

Welcome to release `v1.78.0` of Immich. This release packs a ton of improvement that we hope you enjoy. Some of the highlights include

* **Allow self-signed certificate on the mobile app**
Users with self-signed certificates can now enable this option on the mobile app. 

<p align="center">

<img width="400" alt="image" src="https://github.com/immich-app/immich/assets/27055614/612e5428-097b-46f9-8fd5-a3c2b494091b">

<img width="400" alt="image" src="https://github.com/immich-app/immich/assets/27055614/3fcacd8a-df08-49df-8526-2c825ec6f027">
</p>

* **Notable fix**: Fix the issue with iOS background upload not working properly
* **Notable fix**: Better synchronization mechanism on the mobile app - Before this implementation, the mobile app will fetch all asset information from the server to build the timeline. This caused long building time for any instance with many assets. It will now fetch incremental changes since the app was opened last. This mechanism significantly reduces the loading and building of the timeline when reopening the app.

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Mobile
* fix(mobile): Truncate log messages on the logs list page by @ddshd in https://github.com/immich-app/immich/pull/4003
* chore(mobile): build split APKs by @magicedy in https://github.com/immich-app/immich/pull/3940
* fix(mobile): Clean up image details if there is missing info by @ddshd in https://github.com/immich-app/immich/pull/4018
* feat(mobile): efficient asset sync by @fyfrey in https://github.com/immich-app/immich/pull/3945
* fix(mobile): Fix background backup on iOS by @alextran1502 in https://github.com/immich-app/immich/pull/4038
* fix(mobile): images rendered twice due to rebuild by @fyfrey in https://github.com/immich-app/immich/pull/4060
* feat(mobile): allow self-signed certificate on the mobile app by @ddshd in https://github.com/immich-app/immich/pull/4051
* feat(mobile): upload image assets before videos by @shalong-tanwen in https://github.com/immich-app/immich/pull/3958
* chore(mobile): Translations update by @alextran1502 in https://github.com/immich-app/immich/pull/4077

### Server
* fix(server): update asset with tagged people by @jrasm91 in https://github.com/immich-app/immich/pull/4000
* fix(server): regenerate missing person thumbnails by @daniele-athome in https://github.com/immich-app/immich/pull/3970
* feat(server): don't clear console on nest startup by @etnoy in https://github.com/immich-app/immich/pull/4023
* feat(server): tsconfig: don't clear console on nest startup by @etnoy in https://github.com/immich-app/immich/pull/4032
* fix(server): handle failed ML responses by @mertalev in https://github.com/immich-app/immich/pull/4036
* chore(server): Improve test coverage! by @danieldietzler in https://github.com/immich-app/immich/pull/3889
* refactor(server): android motion photos by @jrasm91 in https://github.com/immich-app/immich/pull/3711

### Web
* feat(server,web): server config by @jrasm91 in https://github.com/immich-app/immich/pull/4006
* feat(web): only log http errors in web container by @etnoy in https://github.com/immich-app/immich/pull/4031
* fix(web): disable shortcut when writing text by @martabal in https://github.com/immich-app/immich/pull/4053
* fix(web): album selection modal overflow and alignment by @jrasm91 in https://github.com/immich-app/immich/pull/4059
* refactor(web): disable shortcut when writing by @martabal in https://github.com/immich-app/immich/pull/4057
* fix(web): Timeline scrollbar shown over menu by @danieldietzler in https://github.com/immich-app/immich/pull/4071
* feat(web): load original image when zooming by @lmichelin in https://github.com/immich-app/immich/pull/4040

### Machine Learning
* fix(ml): set higher default worker timeout by @mertalev in https://github.com/immich-app/immich/pull/4007
* fix(ml): load models in separate threads by @mertalev in https://github.com/immich-app/immich/pull/4034

### Others
* chore: skip mobile app build for dependabot by @jrasm91 in https://github.com/immich-app/immich/pull/4001
* chore(docs): Update README.md by @noamfuss in https://github.com/immich-app/immich/pull/4016
* chore(docs): Add french documentation by @Chiogros in https://github.com/immich-app/immich/pull/4010
* chore(docs): Update FAQ.md by @CtrlAltSudo in https://github.com/immich-app/immich/pull/4030
* chore(docs): Point out that the container must be recreated for env changes to take effect  by @danieldietzler in https://github.com/immich-app/immich/pull/4054
* chore(docs): updated ML documentation by @mertalev in https://github.com/immich-app/immich/pull/4063
* chore(docs): by @bo0tzz in https://github.com/immich-app/immich/pull/4064
* chore(deps): bump stumpylog/image-cleaner-action from 0.2.0 to 0.3.0 by @dependabot in https://github.com/immich-app/immich/pull/3997
* chore(deps): bump docker/build-push-action from 4.1.1 to 4.2.0 by @dependabot in https://github.com/immich-app/immich/pull/4027
* chore(deps): bump docker/build-push-action from 4.2.0 to 4.2.1 by @dependabot in https://github.com/immich-app/immich/pull/4055
* chore(deps): bump docker/metadata-action from 4 to 5 by @dependabot in https://github.com/immich-app/immich/pull/4070
* chore(deps): bump docker/login-action from 2 to 3 by @dependabot in https://github.com/immich-app/immich/pull/4069
* chore(deps): bump docker/build-push-action from 4.2.1 to 5.0.0 by @dependabot in https://github.com/immich-app/immich/pull/4068
* chore(deps): bump docker/setup-qemu-action from 2.2.0 to 3.0.0 by @dependabot in https://github.com/immich-app/immich/pull/4067
* chore(deps): bump docker/setup-buildx-action from 2.10.0 to 3.0.0 by @dependabot in https://github.com/immich-app/immich/pull/4066

## New Contributors
* @magicedy made their first contribution in https://github.com/immich-app/immich/pull/3940
* @noamfuss made their first contribution in https://github.com/immich-app/immich/pull/4016
* @CtrlAltSudo made their first contribution in https://github.com/immich-app/immich/pull/4030
* @lmichelin made their first contribution in https://github.com/immich-app/immich/pull/4040

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.77.0...v1.78.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.78.0'>v1.78.0</a>.</em>

**Comment by rithask** [comment]
I was wondering that when we upload photos and videos from iPhone which is synced to iCloud, does the original photos get uploaded to immich or the smaller sized versions which is used to save storage on iPhone gets uploaded to immich? If it's the latter, how do we set it to upload the original photos and videos?

**Reply by alextran1502** [reply]
I haven't used iCloud, so I cannot confirm this behavior. If the files are on the phone's local storage, the original version is uploaded to Immich

**Reply by martinjuhasz** [reply]
when an app requests an image from ios photo library, it will always be the default original. i dont think there is an option to get a scaled down version even.

**Comment by auanasgheps** [comment]
Finally! Thanks for fixing #3741.
I re-generated all thumbnails (missing did not work) and finally I have all my faces again! :)

**Reply by daniele-athome** [reply]
> (missing did not work) 

That's because the job doesn't check for file existance but just that the thumbnailPath column in the table is an empty string. Checking for file existance will be for another PR perhaps :-)

**Comment by DX37** [comment]
Why does the app now need permissions to access music and audio?

![Screenshot_20230914-075749_Permission controller.png](https://github.com/immich-app/immich/assets/12829428/b54d0450-1100-44a1-8577-aa29bad4eb4b)

**Reply by Hunam6** [reply]
You should probably post this as an issue or discuss it in their discord server

**Comment by hoggatt** [comment]
Awesome update! Love the change that loads only what has changed since last app open.
